TDAT: public void test authenticate runtime exception ( ) throws exception {  <NL> instantiator . add connector ( connector name , new mock connector ( null ,  <NL> new exceptional authentication manager ( null ) , null , null ) ) ;  <NL> authentication response response = manager . authenticate (  <NL> connector name , identity ) ;  <NL> assert not null ( response ) ;  <NL> assert false ( response . is valid ( ) ) ;  <NL> assert null ( response . get data ( ) ) ;  <NL> assert null ( response . get groups ( ) ) ;  <NL>  }  COM: <s> test authenticate throws runtime exception </s>